node/task.ts

```diff
@@ -1775,6 +1803,218 @@ export class CodeCoverageEnabler {
     }
 }
 
+//-----------------------------------------------------
+// Logging Commands
+//-----------------------------------------------------
+
+/**
+ * Upload user interested file as additional log information 
+ * to the current timeline record.
+ * 
+ * The file shall be available for download along with task logs.
+ * 
+ * @param path      Path to the file that should be uploaded.
+ * @returns         void
+ */
+export function uploadFile(path: string) {
+    command("task.uploadfile", null, path);
+}
+
+/**
+ * Upload user interested file as additional log information 
+ * to the current timeline record.
+ * 
+ * The file shall be available for download along with task logs.
+ * 
+ * @param containerFolder   Folder that the file will upload to, folder will be created if needed.
+ * @param name              Artifact name.
+ * @param path              Path to the file that should be uploaded.
+ * @returns                 void
+ */
+export function uploadArtifact(containerFolder: string, name: string, path: string) {
+    command("artifact.upload", { "containerfolder": containerFolder, "artifactname": name }, path);
+}
+
+/**
+ * Upload user interested log to build’s container “logs\tool” folder.
+ * 
+ * @param path      Path to the file that should be uploaded.
+ * @returns         void
+ */
+export function uploadBuildLog(path: string) {
+    command("build.uploadlog", null, path);
+}
+
+/**
+ * Update build number for current build.
+ * 
+ * @param value     Value to be assigned as the build number.
+ * @returns         void
+ */
+export function updateBuildNumber(value: string) {
+    command("build.updatebuildnumber", null, value);
+}
+
+/**
+ * Add a tag for current build.
+ * 
+ * @param value     Tag value.
+ * @returns         void
+ */
+export function addBuildTag(value: string) {
+    command("build.addbuildtag", null, value);
+}
+
+/**
+ * Update release name for current release.
+ * 
+ * @param value     Value to be assigned as the release name.
+ * @returns         void
+ */
+export function updateReleaseName(name: string) {
+    assertAgent("2.132");
+    command("build.updatereleasename", null, name);
+}
+
+/**
+ * Create an artifact link, artifact location is required to be 
+ * a file container path, VC path or UNC share path. 
+ * 
+ * The file shall be available for download along with task logs.
+ * 
+ * @param name              Artifact name.
+ * @param path              Path to the file that should be associated.
+ * @param artifactType      ArtifactType enum of Container, FilePath, VersionControl, GitRef or TfvcLabel.
+ * @returns                 void
+ */
+export function associateArtifact(name: string, path: string, artifactType: ArtifactType) {
+    command("artifact.associate", { "type": artifactType, "artifactname": name }, path);
```

Comment:
Nit: It would probably be good to group these functions by type. So everything with the form ```artifact.x``` goes together, everything with the form ```task.x``` goes together, etc...

Reply:
Group in the code in kind of sections or in classes that are then exported in the task module?

Reply:
Just in sections. Mostly just a style thing to make functions easier to find